src/modules/cdp/diameter_peer.c

```diff
@@ -236,7 +236,7 @@ int diameter_peer_start(int blocking)
 	int seed;
 	peer *p;
 
-	seed = random();
+	seed = kam_rand();
```

Comment:
It looks like this seed is later used to seed the glibc random number generator for (sub-) processes? If yes, then you probably want to use cryptorand() for it. This is an internal kamailio function that returns a cryptographically secure random number. The kam_rand() is just a define for rand() from the glibc, which is quite weak security wise.

This applies also to the other places below.

Please also have a look if you are not seeding the main process again, as this is already done in main.c and core/pt.c

Reply:
It depends what the child processes actually use then in their code. The code in pt.c seeds all three as the different modules uses a lot of different functions. This is usually not needed in individual module new processes.
My other point was that the change here does not change anything regarding the functionality, besides silence a warning. Of course its still better to use the kam_rand() as a policy, to allow easier auditing or later changes.

```
core/rand/kam_rand.h
31:#define kam_rand(x) rand(x)
```

Reply:
Right, that function is actually named `fork_process()`, exactly what I needed :rofl:

Reply:
Seems like that function is already called, so I can probably just remove the seeding from here, since it was already seeded (by calling `fork_process()`).

Reply:
Great, if you are using fork_process from the core this is indeed not needed and it should be removed.